lib/linux_admin/network_interface/network_interface_rh.rb

```diff
@@ -0,0 +1,101 @@
+require 'ipaddr'
+require 'pathname'
+
+module LinuxAdmin
+  class NetworkInterfaceRH < NetworkInterface
+    IFACE_DIR = "/etc/sysconfig/network-scripts"
+
+    # @return [Hash<String, String>] Key value mappings in the interface file
+    attr_reader :interface_conf
+
+    # @param [String] Name of the network interface to manage
+    def initialize(interface)
+      super
+      @iface_file = Pathname.new(IFACE_DIR).join("ifcfg-#{@interface}")
+      parse_conf
+    end
+
+    # Parses the interface configuration file into the @interface_conf hash
+    def parse_conf
+      @interface_conf = {"NM_CONTROLLED" => "no"}
+      contents = File.read(@iface_file)
+
+      contents.each_line do |line|
+        next if line =~ /^\s*#/
+
+        pair = line.split('=').collect(&:strip)
+        @interface_conf[pair[0]] = pair[1]
+      end
+    end
+
+    # Set the IPv4 address for this interface
+    #
+    # @param address [String]
+    # @raise ArgumentError if the address is not formatted properly
+    def address=(address)
+      validate_ip(address)
+      @interface_conf["BOOTPROTO"] = "static"
+      @interface_conf["IPADDR"]    = address
+    end
+
+    # Set the IPv4 gateway address for this interface
+    #
+    # @param address [String]
+    # @raise ArgumentError if the address is not formatted properly
+    def gateway=(address)
+      validate_ip(address)
+      @interface_conf["GATEWAY"] = address
+    end
+
+    # Set the IPv4 sub-net mask for this interface
+    #
+    # @param address [String]
+    # @raise ArgumentError if the mask is not formatted properly
+    def netmask=(mask)
+      validate_ip(mask)
+      @interface_conf["NETMASK"] = mask
+    end
+
+    # Sets one or both DNS servers for this network interface
+    #
+    # @param servers Array<String> The DNS servers
+    def dns=(servers)
+      @interface_conf["DNS1"] = servers[0]
+      @interface_conf["DNS2"] = servers[1] if servers[1]
+    end
+
+    # Sets the search domain list for this network interface
+    #
+    # @param domains [Array<String>] the list of search domains
+    def search_order=(domains)
+      @interface_conf["DOMAIN"] = "\"#{domains.join(' ')}\""
+    end
+
+    # Set up the interface to use DHCP
+    # Removes any previously set static networking information
+    def enable_dhcp
```

Comment:
@bdunne Should we also clear the DNS server and search order here and assume that will be set by DHCP?